Remove can_run_adhoc_query

ranquild opened the pull request:
We added `can_run_adhoc_query` as performance optimization, but it made things worse. We also had a bug where `can_run_adhoc_query` had a different value than `isEditable`. And it made things harder to reason about because there are 2 properties. I think it was a failed experiment, let's revert this change now.

ranquild committed:
Remove can_run_adhoc_query from the FE

nemanjaglumac reviewed (approved):
Code-wise looks clean and straight forward.
I haven't tested the changes locally.

ranquild committed:
Remove can_run_adhoc_query from the FE

dpsutton committed:
Disable can_run_adhoc_query on dashboard

Was impacting performance on dashboard load. Timings locally for me:

get dash: 3178 -> 199
get metadata: 3604 -> 1002

So dashboard went from 6 seconds to load to 1 second.

dpsutton committed:
Fixup test from disabling can_run_adhoc_query

ranquild committed:
Merge branch 'master' into perf-adhoc-query

ranquild committed:
Merge branch 'master' into perf-adhoc-query

perivamsi commented:
resolves https://github.com/metabase/metabase/issues/44359
